
### Examining racial discrimination in the US job market

#### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

#### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes.

#### Exercise
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Discuss statistical significance.

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet

****

In [3]:
import pandas as pd
import numpy as np
from scipy import stats

In [4]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [4]:
# number of callbacks for balck-sounding names
sum(data[data.race=='b'].call)

157.0

In [5]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

In [8]:
data.columns.values.tolist()

['id',
 'ad',
 'education',
 'ofjobs',
 'yearsexp',
 'honors',
 'volunteer',
 'military',
 'empholes',
 'occupspecific',
 'occupbroad',
 'workinschool',
 'email',
 'computerskills',
 'specialskills',
 'firstname',
 'sex',
 'race',
 'h',
 'l',
 'call',
 'city',
 'kind',
 'adid',
 'fracblack',
 'fracwhite',
 'lmedhhinc',
 'fracdropout',
 'fraccolp',
 'linc',
 'col',
 'expminreq',
 'schoolreq',
 'eoe',
 'parent_sales',
 'parent_emp',
 'branch_sales',
 'branch_emp',
 'fed',
 'fracblack_empzip',
 'fracwhite_empzip',
 'lmedhhinc_empzip',
 'fracdropout_empzip',
 'fraccolp_empzip',
 'linc_empzip',
 'manager',
 'supervisor',
 'secretary',
 'offsupport',
 'salesrep',
 'retailsales',
 'req',
 'expreq',
 'comreq',
 'educreq',
 'compreq',
 'orgreq',
 'manuf',
 'transcom',
 'bankreal',
 'trade',
 'busservice',
 'othservice',
 'missind',
 'ownership']

In [9]:
df = data[['race', 'call']]
df.head()

race  call
0    w   0.0
1    w   0.0
2    b   0.0
3    b   0.0
4    w   0.0

In [10]:
df

race  call
0       w   0.0
1       w   0.0
2       b   0.0
3       b   0.0
4       w   0.0
5       w   0.0
6       w   0.0
7       b   0.0
8       b   0.0
9       b   0.0
10      b   0.0
11      w   0.0
12      b   0.0
13      w   0.0
14      b   0.0
15      w   0.0
16      w   0.0
17      b   0.0
18      w   0.0
19      b   0.0
20      b   0.0
21      w   0.0
22      w   0.0
23      w   0.0
24      w   0.0
25      b   0.0
26      b   0.0
27      w   0.0
28      b   0.0
29      b   0.0
...   ...   ...
4840    b   0.0
4841    b   0.0
4842    b   0.0
4843    w   1.0
4844    b   0.0
4845    w   0.0
4846    w   1.0
4847    w   1.0
4848    b   1.0
4849    b   0.0
4850    b   0.0
4851    w   0.0
4852    w   0.0
4853    b   0.0
4854    w   0.0
4855    w   0.0
4856    b   0.0
4857    b   0.0
4858    b   0.0
4859    b   1.0
4860    w   0.0
4861    w   1.0
4862    w   0.0
4863    w   0.0
4864    b   0.0
4865    b   0.0
4866    b   0.0
4867    w   0.0
4868    b   0.0
4869    w   0.0

[4870 rows x 2 columns]

In [11]:
# Separating the dataframe into black and white
df_b = df[df.race == 'b']
df_w = df[df.race == 'w']
# Checking 'n' of each sample
len(df_b), len(df_w)

(2435, 2435)

(0.064476386036960986, 0.096509240246406572)

In [16]:
 sum(df_b.call)

157.0

In [17]:
len(df_b.call)

2435

In [18]:
 sum(df_w.call)

235.0

In [19]:
 len(df_w.call)

2435

In [20]:
# Calculating the proportions per racial group 
prop_b = ((sum(df_b.call))/(len(df_b.call)))
prop_w = ((sum(df_w.call))/(len(df_w.call)))
prop_b, prop_w # p-hats for both black and white

(0.064476386036960986, 0.096509240246406572)

## Question 1 - What test is appropriate for this problem? Does CLT apply?

A hypothesis z-test is appropriate in this instance as we are comparing the observed difference between two proportions (a point estimate) to the null (no difference) and because n > 30. Categorical data differs from numerical data in that we use the sum of 'successes' divided by the 'n' of the sample instead of calculating a mean. In this way we can determine if the proportion of sucesses/failures for each group in the explanatory variable. We can then compute the z-test.
The Central Limit theorum does apply in categorical data for the sampling distribution of the sample proportion. Distribution is nearly normal.

## Question 2 - What are the null and alternate hypotheses?

**Null Hypothesis : H0 : 
No significant differenece between the proportion of calls recieved from CV's with white-sounding names and those with black-sounding names

**Alternative Hypothesis : H1: 
There is a significant difference between the proportion of calls recieved from CV's with white-sounding names and the proportion of those with black-sounding names.

## Compute margin of error, confidence interval, and p-value.

In [21]:
# Finding standard error for proportions (different for hypothesis test)
se_CI = np.sqrt((prop_b*(1 - prop_b)/(len(df_b))) + (prop_w*(1 - prop_w) /(len(df_w))))
se_CI

0.0077833705866767544

In [23]:
# We are calculating at the 5% significance level so our critical value is 1.96
crit = 1.96
margin = abs(crit*se_CI)
print("The true population proportion lies +/- %0.6F around the point estimate" % margin)

The true population proportion lies +/- 0.015255 around the point estimate


In [24]:
# Confidence Interval
#Race is our explanatory varible and Call is our response variable
# Observed Difference
point_est = prop_w - prop_b
point_est

0.032032854209445585

In [25]:
CI = [ point_est + margin, point_est - margin]
print("The proportion is %0.6F +/- %0.5F " % (point_est,margin))
print("The proportion of CVs with white-sounding names that recieve a call is between %0.6F and %0.6F higher than the proportion of CVS with black-sounding names" % (CI[1],CI[0]))

The proportion is 0.032033 +/- 0.01526 
The proportion of CVs with white-sounding names that recieve a call is between 0.016777 and 0.047288 higher than the proportion of CVS with black-sounding names


In [26]:
## Compute P-value
# Using test statistics

# Setting the parameters assuming Ho is true
# Expected Difference of prop_w - prop_b
null = 0

#Create a pooled proportion as expected value of calls across black & white 
p_pool = (sum(data.call)/(len(data.call)))
p_pool

0.080492813141683772

In [27]:
# Calculating Standard Error using the pooled proportion (again assuming that H0 is true)
se_ht = np.sqrt((p_pool*(1 - p_pool)/(len(df_b))) + (p_pool*(1 - p_pool) /(len(df_w))))
se_ht

0.0077968940361704568

In [28]:
# z-score is the observed (prop_w - prop_b) - expected (null) divided by standard error (pooled, computed for the CI above)
# Expected Difference of prop_w - prop_b

z = (point_est - null)/se_ht #standard error calculated in CI above
p_values = stats.norm.sf(abs(z))*2 #twoside
print("Z-score is equal to : %6.3F  p-value equal to: %6.7F" % (z,p_values))

Z-score is equal to :  4.108  p-value equal to: 0.0000398


Z-score is equal to :  4.108  p-value equal to: 0.0000398
As you can see above, we have a p-value of 0.0000398 < 0.025, and as such we can reject the null hypothesis and conclude that there is a significant difference between black-sounding and white-sounding names in the number of calls received.
Just for the sake of diligence, let's double check this answer using the z-test function from Statsmodels

In [29]:
from statsmodels.stats.weightstats import ztest
ztest = ztest(data.call[data.race == 'w'],data.call[data.race == 'b'], alternative = 'two-sided')
print("Z-score is equal to : %6.3F  p-value equal to: %6.7F" % (abs(ztest[0]),ztest[1]))

Z-score is equal to :  4.115  p-value equal to: 0.0000388


We get an almost identical result in both cases. There is a p-value of 0.0000388 < 0.025, and as such we can reject the null hypothesis and conclude that there is a significant difference between black-sounding and white-sounding names in the number of calls received.

## Discuss statistical significance.


If the null hypothesis were true there would only be a 0.00388% chance of the observed data being sampled. This gives strong evidence that there is indeed a statistically significant difference between the two groups based on race. The significance of the finding is strong as it is considerably below the threshold.